aea/helpers/dialogue/base.py

```diff
@@ -132,48 +134,153 @@ def __str__(self):
         )
 
 
-class Dialogue:
+class Dialogue(ABC):
     """The dialogue class maintains state of a dialogue and manages it."""
 
-    def __init__(self, dialogue_label: DialogueLabel) -> None:
+    STARTING_MESSAGE_ID = 1
+    STARTING_TARGET = 0
+
+    class Role(Enum):
+        """This class defines the agent's role in a dialogue."""
+
+        def __str__(self):
+            """Get the string representation."""
+            return self.value
+
+    class EndState(Enum):
+        """This class defines the end states of a dialogue."""
+
+        def __str__(self):
+            """Get the string representation."""
+            return self.value
+
+    def __init__(self, dialogue_label: DialogueLabel, role: Role) -> None:
```

Comment:
For backwards compatibility the role needs to have a default value set. If we do `role: Optional[Role] = None` we will have to take care of the optionality of this field ...alternatively, you could have a dummy role, like `EmptyRole` which is declared here, so `role: Role = EmptyRole(None)` where 
```
class EmptyRole(Role):
    EMPTY = None
```

Not sure what is best. Maybe there is a neater approach as well...

Reply:
Remove if not needed.

Reply:
It is needed. It replaces `is_seller` in the fipa dialogue constructor which was moved up to the base class. 

Done